### ISSUE TITLE
Can't use Vim text-object tag in TSX file

### ISSUE THREAD

**Issue by nauvalazhar** [zed-industries/zed#16722]
### Check for existing issues

- [x] Completed

### Describe the bug / provide steps to reproduce it

Hi, I can't use text-object tags in TSX files like `cit`, `dit`, etc. It works fine in HTML tho.

Steps to reproduce:
1. Create a new file with TSX language mode with the following code:
```tsx
export function Component() {
  return (
    <div>
      <h1>Component</h1>
      <p>Component content</p>
    </div>
  );
}
```
2. Put the cursor on, for example, `<h1>`, and type `cit`.

Thank you.

### Environment

Zed: v0.149.3 (Zed)
OS: macOS 14.2.0
Memory: 8 GiB
Architecture: aarch64

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by hekmyr** [comment]
Hi, I spent a little bit of time trying to understand what is going on, here is what I have found.

* I was not able to reproduce this behavior on commit `fd4a412` (commit right before `596ee58`)
* I was able to reproduce this behavior on commit `596ee58` (commit right after `fd4a412`)
* The source of this behavior seems to bo related to version `0.21.0` of `tree-sitter-typescript`
* The function `surrounding_html_tag()` of the `object.rs` file affected

Given a  TSX file with the following content:

```typescript
export function Component() {
  return (
    <div>
      <h1>Component</h1>
      <p>Component content</p>
    </div>
  );
}
```

And the these values provided by the zed syntax tree:

```
open_tag: jsx_opening_element [4:7-4:11]
  "<": [4:7 - 4:8]
  ">": [4:10 - 4:11]
```

```
close_tag: jsx_closing_element [4:20-4:25]
  "</": [4:20 - 4:22]
  ">": [4:24 - 4:25]
```

* The `parent()` function the `Node` object return unexpected values when the language selected is TSX
   * `{Node jsx_opening_element (2, 9) - (3, 10)}` instead of `{Node jsx_opening_element (3, 6) - (3, 10)}`
   * `{Node jsx_element (3, 6) - (3, 24)}` instead of `{Node jsx_element (2, 9) - (3, 24)}`
* Every coordinate from the `Node` should be incremented by one so `(2, 9)` becomes `[3:10]` and `(3, 6)` becomes `[4:7]` and these values should match the opening and closing tag

**Comment by nicoalbanese** [comment]
this is resolved for me in the latest release (154.2)

**Comment by nauvalazhar** [comment]
> this is resolved for me in the latest release (154.2)

Yeah, I can confirm this has been solved!
